### PR TITLE
Redesign profile page

### LINKED ISSUE #387: Redesigning the Profile Page
## Issue

- We will be following a grid-based system to design a website.
  1. 12 for desktops
  2. 8 for tablets
  3. 2 for phones
- Redesigning profile page to match the grid system.

## Current Profile Page

https://my.realdevsquad.com/profile

204078768-d3c71a9f-075d-4bac-a2d2-62ca06491e74
204078775-b8ecb4ce-9f83-488e-aa4c-e254a3a014d4

## Figma Design

https://www.figma.com/file/72WW0LiRF2s9bwa98hxvlI/Profile-Page?node-id=0%3A1&t=KRzzj3A649JHaXkn-1

### PR DESCRIPTION
**Pull request by manish591**
### What is the change?

- Redesigned the profile page

### Is Development Tested?

- [x] Yes
- [ ] No

### Before / After Change Screenshots

Before :
Image/Video

229308639-fc35b47e-a9f0-4a68-afd1-cacc9d35f239
229308646-ede022a0-c546-4080-bbcc-3906f8b78938

After :
Image/Video

screenshot-localhost_3443-2023 05 17-01_46_37

screenshot-localhost_3443-2023 05 17-01_47_18

### CODE REVIEW

File: app/controllers/profile.js [hunk 1]
```diff
@@ -14,34 +15,176 @@ export default class ProfileController extends Controller {
...
+    {
+      id: 'email',
+      label: 'Email*',
+      type: 'email',
+      required: true,
+      placeholder: 'e.g johndoe@gmail.com',
+      icon_url: 'icons/mail.svg',
+      showError: false,
+      errorMessage: 'Valid email is required',
+      validator: this.emailValidator,
```
Comment (rohan09-raj) [inline app/controllers/profile.js:61]: Email is getting updated from here but doesn't comes with the API, therefore we can't prefill the information here.
image

Can we please confirm that do backend have the email field that is getting added to DB and if it's there, can we send it in the `/users/self` API to  display it here ??

File: app/controllers/profile.js [hunk 2]
```diff
@@ -14,34 +15,176 @@ export default class ProfileController extends Controller {
...
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+
+      const { status } = response;
+      if (status !== 204) {
+        this.toast.error(
+          'Something went wrong. Please check console errors.',
```
Comment (rohan09-raj) [inline app/controllers/profile.js:177]: Can we please display the toast on success with timeout of 1 second and whatever the default is present in codebase as well ??
**Content :** Updated details successfully

Comment (rohan09-raj) [reply app/controllers/profile.js:177]: This leads to one more problem that whenever I want to update any details, due to validations being put on email `Submit` button doesn't get enabled until the email is correctly filled

[Screencast from 2023-05-11 00-37-54.webm](https://github.com/Real-Dev-Squad/website-my/assets/78433013/45510bdd-5729-4041-93c0-02231d5b55ab)

File: app/controllers/profile.js [hunk 3]
```diff
@@ -14,34 +15,176 @@ export default class ProfileController extends Controller {
...
+          '',
+          toastNotificationTimeoutOptions
+        );
+      }
+    } catch (error) {
+      console.error('Error : ', error);
+    }
+    //finally {}
```
Comment (rohan09-raj) [inline app/controllers/profile.js:185]: Can we please remove this comment if we aren't using the finally block here?

File: app/styles/profile.css [hunk 4]
```diff
@@ -1,58 +1,114 @@
...
+  padding-bottom: 2rem;
+}
+
+.profile-field-input-container {
+  display: grid;
+  grid-template-columns: auto 1fr auto;
+  align-items: center;
+  border-radius: 8px;
+  border: 1px solid #d1d5db;
+  background-color: #f9fafb;
```
Comment (rohan09-raj) [inline app/styles/profile.css:34]: Can we please CSS variables for the color, there should be a file present which contains all the variables ?

File: app/styles/profile.css [hunk 5]
```diff
@@ -1,58 +1,114 @@
...
+
+.profile-field-input {
+  padding: 0 1.25rem;
+  border: none;
+  background: none;
+  outline: none;
+}
+
+.profile-field-input:active {
+  border-color: #1c64f2;
```
Comment (rohan09-raj) [inline app/styles/profile.css:50]: Re: Use CSS variables

File: app/styles/profile.css [hunk 6]
```diff
@@ -1,58 +1,114 @@
...
+  grid-template-columns: auto 1fr auto;
+  align-items: center;
+  border-radius: 8px;
+  border: 1px solid #d1d5db;
+  background-color: #f9fafb;
+  padding: 1rem;
 }
+
+.profile-field-input-container:focus-within {
+  border-color: #1c64f2;
```
Comment (rohan09-raj) [inline app/styles/profile.css:39]: Re: Use CSS variables

File: app/styles/profile.css
Comment (rohan09-raj) [inline app/styles/profile.css:1]: Can we please check why pointer cursor is coming on the whole page?

File: app/styles/profile.css [hunk 7]
```diff
@@ -1,58 +1,114 @@
...
+  cursor: pointer;
+  display: none;
+}
+
+.profile-field-input-delete:nth-of-type(2) {
+  display: block;
+}
+
+.profile-field-error-message {
+  color: red;
```
Comment (rohan09-raj) [inline app/styles/profile.css:65]: Re: Use CSS variables

File: app/styles/profile.css [hunk 8]
```diff
@@ -1,58 +1,114 @@
...
+
 .edit-btn {
-  position: absolute;
-  right: -10px;
-  top: -10px;
-  font-size: 1.6em;
+  padding: 0;
+  font-size: 1rem;
+  margin-top: 0.5rem;
+  color: #1e429f;
```
Comment (rohan09-raj) [inline app/styles/profile.css:102]: Re: Use CSS variables

File: app/styles/profile.css [hunk 9]
```diff
@@ -77,6 +133,25 @@
   min-width: 400px;
 }
 
+.profile-form-submit {
+  border-radius: 8px;
+  border: none;
+  background-color: #1c64f2;
```
Comment (rohan09-raj) [inline app/styles/profile.css:139]: Re: Use CSS variables

File: app/styles/profile.css [hunk 10]
```diff
@@ -77,6 +133,25 @@
...
+  padding: 1rem 4rem;
+}
+
+.profile-form-submit:disabled {
+  background-color: #e5efeb;
+  color: #9ca3af;
+}
+
+.profile-field-error .profile-field-input-container {
+  border-color: red;
```
Comment (rohan09-raj) [inline app/styles/profile.css:152]: Re: Use CSS variables

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: app/controllers/profile.js (hunk 1 above)
Comment (manish591) [reply app/controllers/profile.js:61]: ok, Will confirm.

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**